### PR TITLE
Added new sample demonstrating launching activities

### PR DESCRIPTION
**Pull request by makaroffandrey**
**Description**: Created a sample that demonstrates how to launch a non RIBs activity from a RIB and receive the result back. The sample was created by extracting similar logic from the sandbox app.

### CODE REVIEW

File: samples/android-integration/launching-activities/src/main/java/com/badoo/ribs/samples/android_integration/launching_activities/rib/LaunchingActivitiesView.kt [hunk 1]
```diff
@@ -17,7 +18,14 @@ import com.badoo.ribs.samples.android_integration.launching_activities.R
 
 interface LaunchingActivitiesView : RibView {
 
-    interface Factory : ViewFactoryBuilder<Nothing?, LaunchingActivitiesView>
+    interface Dependency {
+        @StringRes
+        fun getTitleResource(): Int
```
Comment (zsoltk) [inline samples/android-integration/launching-activities/src/main/java/com/badoo/ribs/samples/android_integration/launching_activities/rib/LaunchingActivitiesView.kt:23]: Let's use `val`s

File: samples/android-integration/launching-activities/src/main/java/com/badoo/ribs/samples/android_integration/launching_activities/rib/builder/LaunchingActivitiesChildBuilder.kt [hunk 2]
```diff
@@ -2,22 +2,29 @@ package com.badoo.ribs.samples.android_integration.launching_activities.rib.buil
...
         )
+        val viewDependency = object : LaunchingActivitiesView.Dependency {
+            override fun getTitleResource(): Int = R.string.child_title
+            override fun getDescriptionResource(): Int = R.string.child_description
+        }
+
         return LaunchingActivitiesNode(
                 buildParams = buildParams,
-                viewFactory = LaunchingActivitiesViewImpl.Factory().invoke(null),
+                viewFactory = LaunchingActivitiesViewImpl.Factory(layoutRes = R.layout.rib_launching_activities).invoke(viewDependency),
```
Comment (zsoltk) [inline samples/android-integration/launching-activities/src/main/java/com/badoo/ribs/samples/android_integration/launching_activities/rib/builder/LaunchingActivitiesChildBuilder.kt:27]: Don't override default layout here. If anywhere, then in customisation, but we don't have that in this sample (which is alright).

For the same reason, maybe extract the view factory instance itself to a field, as it usually comes from customisation, not created on the fly.

File: samples/android-integration/launching-activities/src/main/java/com/badoo/ribs/samples/android_integration/launching_activities/rib/builder/LaunchingActivitiesChildBuilders.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.badoo.ribs.samples.android_integration.launching_activities.rib.builder
```
Comment (zsoltk) [inline samples/android-integration/launching-activities/src/main/java/com/badoo/ribs/samples/android_integration/launching_activities/rib/builder/LaunchingActivitiesChildBuilders.kt:1]: Package structure seems merged.

It should be:
`.rib` (a package to hold all ribs defined in this module, each in its own subpackage)
`.rib.parent` (A single rib)
`.rib.child` (Another rib)
and no `.builder` subpackage if it only contains a single file.

Comment (makaroffandrey) [reply samples/android-integration/launching-activities/src/main/java/com/badoo/ribs/samples/android_integration/launching_activities/rib/builder/LaunchingActivitiesChildBuilders.kt]: It's not the package structure that is merged. I used the same node class for both parent and child.
I have changed it. Now it is a parent that hosts two children. The new package structure closely resembles the routing example.

File: samples/android-integration/launching-activities/src/main/java/com/badoo/ribs/samples/android_integration/launching_activities/rib/LaunchingActivitiesView.kt (hunk 1 above)
Comment (makaroffandrey) [reply samples/android-integration/launching-activities/src/main/java/com/badoo/ribs/samples/android_integration/launching_activities/rib/LaunchingActivitiesView.kt]: Done.

File: samples/android-integration/launching-activities/src/main/java/com/badoo/ribs/samples/android_integration/launching_activities/rib/builder/LaunchingActivitiesChildBuilder.kt (hunk 2 above)
Comment (makaroffandrey) [reply samples/android-integration/launching-activities/src/main/java/com/badoo/ribs/samples/android_integration/launching_activities/rib/builder/LaunchingActivitiesChildBuilder.kt]: I have changed the rib hierarchy so overriding layouts is not needed anymore.

**Review by zsoltk — APPROVED** [review]

**Pull request merged**